ql/src/java/org/apache/hadoop/hive/ql/plan/Statistics.java

```diff
@@ -220,14 +220,14 @@ public Statistics clone() {
   }
 
   public void addBasicStats(Statistics stats) {
-    dataSize += stats.dataSize;
+    dataSize = StatsUtils.safeAdd(dataSize, stats.dataSize);
     numRows += stats.numRows;
```

Comment:
Can we also use `safeAdd` here? [This line in the same file](https://github.com/apache/hive/blob/4654aea51b3369d7845ab129ece496672979fee9/ql/src/java/org/apache/hadoop/hive/ql/plan/Statistics.java#L258) does have a possibility of overflow. I know # of rows is unlikely to be overflowed, though.

Reply:
You are right, thank you. I was also hesitant about this. With your suggestion, it does seem more appropriate to add it.

Reply:
Sorry, could you please update also this line? With it, I think this class will be perfect and no one will have to touch it anymore.

```
updatedCS.setNumNulls(updatedCS.getNumNulls() + cs.getNumNulls());
```

Reply:
sorry，please forgive my unfamiliarity with GitHub. I will make the changes to this file right away.